service/controller/chart/v1/resource/tillermigration/create.go

```diff
@@ -0,0 +1,100 @@
+package tillermigration
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/chart-operator/service/controller/chart/v1/key"
+)
+
+// EnsureCreated ensures tiller resources are deleted after all helm releases migrated into v3.
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Resource is used to remove tiller pod in tenant clusters.
+	// So for other charts we can skip this step.
+	if key.ReleaseName(cr) != key.ChartOperatorName {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no need to delete a tiller for %#q", key.ReleaseName(cr)))
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		return nil
+	}
+
+	charts, err := r.g8sClient.ApplicationV1alpha1().Charts("giantswarm").List(metav1.ListOptions{})
```

Comment:
```suggestion
	charts, err := r.g8sClient.ApplicationV1alpha1().Charts("").List(metav1.ListOptions{})
```

I don't really like the hardcoded namespace here.